scikeras/_utils.py

```diff
@@ -1,60 +1,66 @@
 import inspect
 import os
 import random
-import warnings
 
+from contextlib import contextmanager
 from types import FunctionType
-from typing import Any, Callable, Dict, Iterable, Mapping, Sequence, Type, Union
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Generator,
+    Iterable,
+    Mapping,
+    Sequence,
+    Type,
+    Union,
+)
 
 import numpy as np
 import tensorflow as tf
 
-from numpy.lib.arraysetops import isin
 from tensorflow.keras import losses as losses_mod
 from tensorflow.keras import metrics as metrics_mod
 from tensorflow.keras import optimizers as optimizers_mod
+from tensorflow.python.eager import context
+from tensorflow.python.framework import config, ops
 
 
 DIGITS = frozenset(str(i) for i in range(10))
 
 
-class TFRandomState:
```

Comment:
Why rename `TFRandomState`?

Reply:
We don't need to rename it. I just did it while I was in the flow of reworking it to fix the actual issues. It is private, so in theory it doesn't hurt anything other than polluting this diff. I do like this style more, so I don't think it's worth it to revert just for the diff. Apologies for the confusion.

Reply:
If it's private, I would recommend removing it from the docs:

https://github.com/adriangb/scikeras/blob/67de2388f34eecd1f09d4b1deaeb8ada5c60dc1b/docs/source/notebooks/Benchmarks.md#L109

Reply:
I agree it should not be in the docs. But let's table that for a future discussion.

Reply:
Opened #253 to track